src/main/java/us/kbase/auth2/lib/Authentication.java

```diff
@@ -2339,7 +2353,8 @@ public NewToken login(
 						linked, u.get().getUserName().getName());
 			}
 		}
-		return login(u.get().getUserName(), tokenCtx);
+		final MfaStatus mfaStatus = ri.get().getDetails().getMfa();
+		return login(u.get().getUserName(), tokenCtx, mfaStatus);
```

Comment:
I don't understand how this can ever work - the MFA status is coming from the remote identity stored as part of the temporary session data, but the MFA status isn't persisted to the temporary session data in mongo

Reply:
:+1: